docs/proposals/20201020-management-cluster-operator.md

```diff
@@ -0,0 +1,844 @@
+
+* [Management cluster operator](#management-cluster-operator)
+  * [Table of Contents](#table-of-contents)
+  * [Glossary](#glossary)
+  * [Motivation](#motivation)
+     * [Goals](#goals)
+     * [Non-Goals/Future Work](#non-goalsfuture-work)
+  * [Proposal](#proposal)
+     * [User Stories](#user-stories)
+     * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+        * [Existing API Types Changes](#existing-api-types-changes)
+        * [New API Types](#new-api-types)
+        * [Example API Usage](#example-api-usage)
+        * [Operator Behaviors](#operator-behaviors)
+           * [Installing a provider](#installing-a-provider)
+           * [Upgrading a provider](#upgrading-a-provider)
+           * [Upgrades providers without changing contract](#upgrades-providers-without-changing-contract)
+           * [Upgrades providers and changing contract](#upgrades-providers-and-changing-contract)
+           * [Changing a provider](#changing-a-provider)
+           * [Deleting a provider](#deleting-a-provider)
+        * [Operator Lifecycle Management](#operator-lifecycle-management)
+           * [Operator Installation](#operator-installation)
+           * [Operator Upgrade](#operator-upgrade)
+           * [Operator Delete](#operator-delete)
+        * [Air gapped environment](#air-gapped-environment)
+     * [Risks and Mitigation](#risks-and-mitigation)
+        * [Upgrade from v1alpha3 management cluster to v1apha4/operator cluster](#upgrade-from-v1alpha3-management-cluster-to-v1apha4operator-cluster)
+  * [Additional Details](#additional-details)
+     * [Test Plan](#test-plan)
+     * [Version Skew Strategy](#version-skew-strategy)
+  * [Implementation History](#implementation-history)
+
+## Glossary
+
+The lexicon used in this document is described in more detail
+[here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/glossary.md).
+Any discrepancies should be rectified in the main Cluster API glossary.
+
+## Summary
+
+The clusterctl CLI currently handles the lifecycle of Cluster API
+providers installed in a management cluster. It provides a great Day 1
+experience in getting CAPI up and running. However, clusterctl’s imperative
+design could make it difficult for cluster admins to stand up and manage CAPI
+management clusters in their own preferred way.
+
+This proposal provides a solution that leverages a declarative API and an
+operator to empower admins to handle the lifecycle of providers within the
+management cluster
+
+## Motivation
+
+In its current form clusterctl is designed to provide a super simple user
+experience for day 1 operations of a Cluster API management cluster with a
+single infrastructure provider.
+
+However such design implicitly is not optimized for supporting declarative
+approaches when operating Cluster API management clusters (e.g. replicating
+clusterctl init is complex).  With the management cluster operator, we aim to
```

Comment:
What about:

However such design does not work well with declarative approaches, because the integration of the clusterctl CLI in a pipeline triggered by git commits could be problematic, and without the CLI the users are losing clusterctl advantages like e.g. the logic for fetching latest version of a provider embedded in clusterctl init.

> What is the word "implicitly" doing here?
> What does it mean to optimize for declarative approaches?

What about ""